targettest/sym1/symNotepad.c

```diff
@@ -0,0 +1,191 @@
+      }
+
+      while( writing ) {
+
+         c = getchar();
+
+         if( c == 0x08 ) {             // Backspace
+            if( p > 0 ) {
+               buffer[p] = 0x00;
+               p--;
+            }
+         }
+         else if( c == 0x13 ) {        // Save
+            puts( "\n========================= Save =========================" );
+            puts( "\nPress any key to save." );
+            c = getchar();
+            for( l = 0; l <= p; l++ ) {
+               tapio[l] = buffer[l];
+            }
+            l++;
+            tapio[l] = 0x00;
+            puts( "Saving to tape." );
+            error = dumpt( 'N', tapio, tapio+p );
+            if( error ) {
+               puts( "\nTape error." );
+            }
+            else
+            {
+               putchar( '\r' );
+               for( l = 0; l < 25; l++ ) {
+                  putchar( '\n' );
+               }
+            }
+            puts( "===================== Sym-1 Notepad ====================\n" );
+            for( l = 0; l <= p; l++ ) {
+               putchar( buffer[l] );
+            }
+         }
+         else if( c == 0x0C ) {        // Load
+            p = 0;
+            puts( "\nLoading from tape." );
+            memset( buffer, 0, heap_size );
+            memset( tapio, 0, TAPIO_MAX_SIZE );
+            error = loadt( 'N' );
+            if( error ) {
+               puts( "\nTape error." );
+               puts( "===================== Sym-1 Notepad ====================\n" );
+            }
+            else
+            {
+               for( l = 0; l <= heap_size; l++ ) {
+                  buffer[l] = tapio[l];
+               }
+
+	       p = strlen( buffer );
+
+               putchar( '\r' );
+               for( l = 0; l < 25; l++ ) {
+                  putchar( '\n' );
+               }
+               puts( "===================== Sym-1 Notepad ====================\n" );
+
+               for( l = 0; l <= p; l++ ) {
+                  putchar( buffer[l] );
+               }
+            }
+         }
+         else if( c == 0x03 ) {        // Clear
+            p = 0;
+            memset( buffer, 0, heap_size );
+            putchar( '\r' );
+            for( l = 0; l < 25; l++ ) {
+               putchar( '\n' );
+            }
+            puts( "===================== Sym-1 Notepad ====================\n" );
+         }
+         else if( c == 0x18 ) {        // Exit
+            writing = 0;
+            running = 0;
+         }
+         else {
+            if( p >= heap_size - 1 ) {
+               puts( "\n========================= End  =========================" );
+               puts( "Buffer full." );
+            }
+            else {
+               if( c == '\n' ) {
+                  putchar( '\n' );
+               }
+               buffer[p] = c;
+            }
+            p++;
+         }
+      }
+   }
+
+   free( buffer );
+
+   puts( "\nEnjoy your day!\n" );
+}
```

Comment:
Please adhere to the cc65 code style, see symDisplay.c.

Reply:
Updated.  Please check and confirm.